dev_docs/07_visual_testing_guide.md

```diff
@@ -0,0 +1,93 @@
+# Visual Testing
+
+The KDS now has a visual testing system integrated into it, using which you can verify the expected visual states after making any changes to the components.
+
+## Implementation details
+
+  The visual testing mechanism uses Puppeteer(for interacting with the testing environment and the rendered components), Jest-Puppeteer(to provide all required configuration to run tests using Puppeteer.), and Percy(to take snapshots for comparing visual diffs) to ensure components render correctly under various conditions. The key components include:
+
+1. **Configuration Files**:
+   - [***visual.index.js***](../jest.conf/visual.index.js): Configures Jest-Puppeteer and includes server checks to ensure the visual testing playground is up and running.
+   - [***visual.setup.js***](../jest.conf/visual.setup.js): Sets up global functions and constants needed for visual tests.
+
+2. **Utility Functions** [***(visual.testUtils.js)***](../jest.conf/visual.testUtils.js):
+   - `renderComponent(component, props, slots)`: Renders the specified component with given props and slots in the visual testing playground.
+   - `takeSnapshot(name, options)`: Takes a Percy snapshot with the given name and options.
+   - `click(selector)`, `hover(selector)`, `scrollToPos(selector, scrollOptions)`, `waitFor(selector)`, `delay(time)`: Utility functions for simulating user interactions.
+
+3. **Visual Testing Playground** ([***testing-playground.vue***](../docs/pages/testing-playground.vue)):
```

Comment:
I think we can go a little deeper, and reframe these 3 points, to explain the idea that we need to run a devserver in parallel with the visual tests, additionally that we use jest to execute the tests (so thats why we need the visual.index.js and visual.setup.js config files), and we use the testing-playground page that is rendered by the devserver to render the components of the visual tests using the mentioned utility funtions. And that to write tests cases we updated the `decribe` and `it` objects to include the `describe.visual` method, and that it adds a `[Visual]` tag to the test name, and we filter by this tag to know which tests should jest execute. So this way the reader can connect the points and realize all the things that are involved, so if they need to make any change to the process, they can understand all the implications.

Reply:
Regarding the changes to `describe` and `it` blocks in case of visual tests, it has already been told in the second point under writing visual tests heading. I can expand upon that a bit more if that's okay.

Reply:
Yes, it would be good explaining that it prepends the "[Visual]" tag and jest filter by that tag. Something simple.